app/pages/ResetPasswordPage.js

```diff
@@ -0,0 +1,119 @@
+import React, { Component } from 'react'
+import { withRouter, Switch, Route } from 'react-router-dom'
+import NotFound from '../components/shared/NotFound'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import formProps from '../components/shared/form/formPropsSelector'
+import RequestResetEmailForm from '../components/ResetPassword/RequestResetEmailForm'
+import ResetPasswordForm from '../components/ResetPassword/ResetPasswordForm'
+import { sendResetPasswordEmail, submitResetPassword, getUserDataFromToken } from '../actions/resetPasswordActions'
+import { setErrorMessage } from '../actions/appActions'
+
+class ResetPasswordPage extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      submitClicked: null
+    }
+  }
+
+  componentWillMount() {
+    const tokenString = this.props.location.pathname.substring(
+      this.props.match.url.length + 1,
+      this.props.location.pathname.length
+    )
+    if (tokenString.length > 0) {
+      this.props.loadInitialData(tokenString)
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.form.validity.passwordsMatch !== nextProps.form.validity.passwordsMatch) {
+      if (!nextProps.form.validity.passwordsMatch) {
+        this.props.passwordsMatchMessage('Passwords do not match.')
+      }
+    }
+  }
+
+  onSubmitClicked = () => {
+    this.setState({
+      submitClicked: new Date().valueOf()
+    })
+  }
+
+  handleResetPasswordSubmit(values) {
+    const { user } = this.props
+    const payload = Object.assign({}, values, user.user)
+    this.props.handleResetPasswordSubmit(payload)
+  }
+
+  render() {
+    const { match, user, model, form, handleSendEmailSubmit } = this.props
+    return (
+      <div className="reset-password-page">
+        <Switch>
+          <Route
+            exact
+            path={match.url}
+            render={() =>
+              <RequestResetEmailForm
+                model={model}
+                form={form}
+                user={user}
+                submitClicked={this.onSubmitClicked}
+                handleSubmit={handleSendEmailSubmit}
+              />}
+          />
+
+          <Route
+            path={`${match.url}/:tokenString`}
+            render={() =>
+              <ResetPasswordForm
+                model={model}
+                form={form}
+                user={user}
+                submitClicked={this.onSubmitClicked}
+                handleSubmit={values => this.handleResetPasswordSubmit(values)}
```

Comment:
why is this function on the class where the other `handleSubmit` was a prop?

Reply:
the function computes form values while the prop is a straight dispatch action